lib/src/license_detection/tokenizer.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:collection';
+import 'dart:convert';
+
+class Token {
+  String token;
+  int? position;
+  final int line;
+
+  Token(this.token, this.position, this.line);
+}
+
+class Tokenizer {
+  static const lineSplitter = LineSplitter();
+  static List<Token> tokenize(String text) {
+    var tokens = <Token>[];
+
+    var lines = lineSplitter.convert(text);
+    var position = 0;
+
+    for (var i = 1; i <= lines.length; i++) {
+      var line = lines[i - 1];
+
+      var buffer = StringBuffer();
+      var prevSignificant = false;
+
+      for (var rune in line.runes) {
```

Comment:
Will this handle uncode correctly? Should consider looking at using: https://pub.dev/packages/characters

Reply:
Yes, it will, the package `Runes` and the `runes` provided by dart are the same. They have mentioned this in their documentation. We use [Grapheme Clusters(Characters)](https://pub.dev/documentation/characters/latest/characters/Characters-class.html) provided them but sin;t it better to work with numbers rather than character?